services/kvstore/src/test/java/tech/pegasys/pantheon/services/kvstore/AbstractKeyValueStorageTest.java

```diff
@@ -62,6 +63,20 @@ public void put() throws Exception {
         Optional.of(BytesValue.fromHexString("0DEF")), store.get(BytesValue.fromHexString("0F")));
   }
 
+  @Test
+  public void mayContainKey() throws Exception {
```

Comment:
Naming of this test is suspicious. There's no "may" in it anymore, we're getting a very explicit response to whether this key exists or not.